h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -1085,9 +1099,61 @@ public void learn() {
           if (!(aModel instanceof StackedEnsembleModel))
             notEnsembles[notEnsembleIndex++] = aModel._key;
 
-        Job<StackedEnsembleModel> ensembleJob = stack(notEnsembles);
+        Job<StackedEnsembleModel> ensembleJob = stack(false, notEnsembles);
         pollAndUpdateProgress(Stage.ModelTraining, "StackedEnsemble build", 50, this.job(), ensembleJob, JobType.ModelBuild);
 
+        //Get top models from each model type (GLM, GBM, DL, DRF, and XRT)
```

Comment:
This code can be generalized so that it can remain unchanged when we eg. add XGboost to the mix or other algos.

you can add a function

String getModelType(Model m): if m._key.toString().startsWith("XRT_") return XRT else return m._parms.algoName;

then you can do

List<Model> models = new ArrayList();
Set<String> types = new HashSet();

for (Model m : allModels) {
   String type = getModelType(m);
   if (types.contains(type) continue;
   types.add(type);
   models.add(m);
}


        Key<Model>[] bestModelKeys = new Key[models.size()];
        int bestModelIndex = 0;
        for (int i = 0; i < models.size(); i++)
            bestModelKeys[i] = models.get(i)._key;

Reply:
Yes, good suggestion. Ty!